docs/development/components-and-directories.md

```diff
@@ -0,0 +1,206 @@
+to be split up into single components as well. **`templates`** might be moved
+completely into the UI-framework once legacy UI components are removed completely.
+**`Customizing`** will be a candidate for future treatment once `lang` and `template`
+are reconsidered.
+
+**`public`** is introduced so installations can stop exposing each and every file
+in the ILIAS repository. This is bad for security and reliability, as the ILILAS
+repository contains loads of files that should not be accessible via web: files
+containing automated tests, build artifacts, documentation.
+
+
+### `components` Folder and Namespacing
+
+The `components` folder is the central folder for the ILIAS development. It contains
+components that are maintained and developed via the processes of the ILIAS society
+and community. It may also contain components developed by third parties, formerly
+known as plugins. Hence, this folder is an implementation of one central aim of
+the [Component Revision 2022](https://docu.ilias.de/goto.php?target=wiki_1357_Component_Revision_2022):
+Mechanisms used by core components and plugins should be aligned.
+
+#### How
+
+The top-level in the `components` starts with a list of all vendors that provide
+components for the installation at hand. Hence, for a standard ILIAS, only one
+folder `ILIAS` will be contained. When the installation uses plugins, the vendors
+of the plugin will have folders as well.
+
+The folders of the individual vendors in turn contain one folder for each component
+provided by the vendor in question. Hence a component folder might look as such:
+
+* ILIAS
+  * AccessControl
+  * Administration
+  * Blog
+  * Course
+  * ...
+* ServiceProvider1
+  * Plugin1
+  * Plugin2
+* ServiceProvider2
+  * Plugin1
+
+Each of these folder contains an according component. The `\src` directory
+in that folder contains the root of an according namespace, as described
+by [PSR-4](https://www.php-fig.org/psr/psr-4/). For example: `components/ILIAS/AccessControl`
+contains the component `ILIAS\AccessControl`, `components/ILIAS/AccessControl/src`
+is the root of the namespace `ILIAS\AccessControl`.
+
+A components folder also contains this substructure:
+
+* `\docs` for component specific documentation
+* `\src` for the code to be used for production
+* `\tests for code that performs automated tests
+* a `$COMPONENT.php` (e.g. AccessControl.php) that defines the binding of the
+  component to the rest of the system
+* a `component.json` that contains component metadata, such as contact of
+  maintainers
+* a `README.md` that gives human readable info about the component
+
+#### Why
+
+PHP-Namespacing with its `Vendor\Package` logic is a good fit for ILIAS components.
+It provides a natural way to resolve naming clashes between all component and
+plugins and fits well in the general PHP ecosystem.
+
+Although for a standard ILIAS installation the additional `ILIAS` folder in `components`
```

Comment:
I see problems with the git ignore here if plugins are just dropped into the "core_components" folder. Its always an uncommitet change in the repo if its not added to the .gitignore. If its added then you always have to specify files you want to commit. 

In Typo3 they choose to split core_coponents and plugin/additional_coponents in different folders e.g. sysext/ext. The ext folder is ignored in git, sysext is not.

Reply:
Interesting point... I think we should be able to solve this via some lesser known syntax for [.gitignore](https://git-scm.com/docs/gitignore):

```
/component/**
!/component/ILIAS
```